### PR TITLE
makes skeleton heads get the typing indicator instead of their body

### PR DESCRIPTION
**Pull request by Skeletonman0**
[QOL]

## About the PR 
This PR aims to make it so decapitated heads from skeletons get the typing indicator instead of their body.

## Why's this needed? 
the head talks, not the body. why does the body get the typing indicator?

talking skeleton head go brrrrrrr

image

### CODE REVIEW

**Comment by Kapu1178** [comment]
The method of doing this is _really snowflakey_. You should use a signal to control this, by sending a signal and looking for a specific response, and if that response is gotten, you assume it was already handled.

File: code/obj/item/organs/head.dm [hunk 1]
```diff
@@ -456,6 +462,10 @@
 
 	attach_organ(var/mob/living/carbon/M as mob, var/mob/user as mob)
 		/* Overrides parent function to handle special case for attaching heads. */
+		if (src.linked_human)
+			src.UnregisterSignal(src.linked_human, "create_typing", .proc/create_typing_indicator)
+			src.UnregisterSignal(src.linked_human, "remove_typing", .proc/remove_typing_indicator)
+			src.UnregisterSignal(src.linked_human, "speech_bubble", .proc/speech_bubble)
```
Comment (Tarmunora) [inline code/obj/item/organs/head.dm:468]:
```suggestion
			src.UnregisterSignal(src.linked_human, COMSIG_CREATE_TYPING)
			src.UnregisterSignal(src.linked_human, COMSIG_REMOVE_TYPING)
			src.UnregisterSignal(src.linked_human, COMSIG_SPEECH_BUBBLE)
```

File: code/obj/item/organs/head.dm [hunk 2]
```diff
@@ -293,6 +295,10 @@
 	on_removal()
 		donor.flags |= OPENCONTAINER
 		src.transplanted = 1
+		if (src.linked_human)
+			src.RegisterSignal(src.linked_human, "create_typing", .proc/create_typing_indicator)
+			src.RegisterSignal(src.linked_human, "remove_typing", .proc/remove_typing_indicator)
+			src.RegisterSignal(src.linked_human, "speech_bubble", .proc/speech_bubble)
```
Comment (Tarmunora) [inline code/obj/item/organs/head.dm:301]:
```suggestion
		if (src.linked_human)
			src.RegisterSignal(src.linked_human, COMSIG_CREATE_TYPING, .proc/create_typing_indicator)
			src.RegisterSignal(src.linked_human, COMSIG_REMOVE_TYPING, .proc/remove_typing_indicator)
			src.RegisterSignal(src.linked_human, COMSIG_SPEECH_BUBBLE, .proc/speech_bubble)
```

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/datums/mutantraces.dm [hunk 3]
```diff
@@ -1140,13 +1140,20 @@ TYPEINFO(/datum/mutantrace)
...
 			var/datum/mutantrace/skeleton/S = H.mutantrace
 			if (H.eye == head)
 				H.set_eye(null)
 			S.head_tracker = null
 			boutput(H, "<span class='alert'><b>You feel as if your head has been repossessed by another!</b></span>")
+		// if we were previously linked to another head
+		if (src.head_tracker)
+			src.head_tracker.UnregisterSignal(src.head_tracker.linked_human, "create_typing", /obj/item/organ/head.proc/create_typing_indicator)
+			src.head_tracker.UnregisterSignal(src.head_tracker.linked_human, "remove_typing", /obj/item/organ/head.proc/remove_typing_indicator)
+			src.head_tracker.UnregisterSignal(src.head_tracker.linked_human, "speech_bubble", /obj/item/organ/head.proc/speech_bubble)
```
[the remark this answers is not in the document]
Comment (flrsh) [reply code/datums/mutantraces.dm:1155]: Oh right, thanks!

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**